app/src/api/services/domainLogic/secondaryMaterial.ts

```diff
@@ -0,0 +1,29 @@
+import _ from 'lodash';
+
+import * as materialDataAccess from '../../dataAccess/material';
+import { ArgumentError } from '../../errors/general';
+
+export interface materialGroup {
+    materials: string[];
+}
+
+export async function updateSecondaryMaterials(update: materialGroup): Promise<materialGroup> {
+    const groupMaterials = update.materials;
+
+    for(let group of groupMaterials) {
+        const isGroupValid = await materialDataAccess.isMaterialAllowed(group);
+        if(!isGroupValid) {
+            throw new ArgumentError(`'${group}' is not a valid Land Use Group`, 'secondaryMaterialsUpdate');
+        }
+    }
+
+    if(hasDuplicates(groupMaterials)) {
+        throw new ArgumentError('Cannot supply duplicate materials', 'secondaryMaterialsUpdate');
```

Comment:
Looks good to me, I have just one comment - perhaps similar to this error thrown here, there can be one for when the Main Secondary Construction Material/s selected is the same as the Core Material (presumably we don't want that)